src/rgw/rgw_op.cc

```diff
@@ -4479,12 +4683,29 @@ void RGWAbortMultipart::execute()
         store->update_gc_chain(meta_obj, obj_part.manifest, &chain);
         RGWObjManifest::obj_iterator oiter = obj_part.manifest.obj_begin();
         if (oiter != obj_part.manifest.obj_end()) {
-          rgw_obj head = oiter.get_location();
+          obj = oiter.get_location();
           rgw_obj_key key;
-          head.get_index_key(&key);
+          obj.get_index_key(&key);
           remove_objs.push_back(key);
         }
       }
+        map<string, bufferlist> attrset;
+        int y = get_obj_attrs(store, s, obj, attrset);
+        map<string, bufferlist>::iterator cmp = attrset.find(RGW_ATTR_COMPRESSION);
+        if (!y && cmp != attrset.end()) {
+          RGWCompressionInfo cs_info;
+          bufferlist::iterator bliter = cmp->second.begin();
+          try {
+            ::decode(cs_info, bliter);
+          } catch (buffer::error& err) {
+            ldout(s->cct, 5) << "Failed to get decompressed obj size" << dendl;
+          }
+          if (cs_info.compression_type != "none")
+            deleted_size += cs_info.orig_size;
```

Comment:
RGWCompressionInfo default constructor initialize compression_type as "none", nothing can happens, as I thought, but it's safer to check it in try, you are right.